docs/access-handler/providers.md

```diff
@@ -0,0 +1,164 @@
+## Providers
+
+The access handler uses a provider framework which allows new services to be integrated easily by implementing a specific interface.
+
+### Adding a provider
+
+Providers are defined in accesshandler/pkg/providers if this provider is for a service which may have multiple possible providers, create a parent folder and
+then create a specific folder for your new provider. E.g aws/sso or azure/ad otherwise, a single folder is ok e.g okta
+
+One of the simplest ways to get started is to copy an existing provider and update the api calls.
+A provider should have the following files.
+
+```
+accesshandler
+ ┗ pkg
+   ┗ providers
+     ┗ okta
+       ┣ 📜 access.go
+       ┣ 📜 errors.go
+       ┣ 📜 okta_test.go
+       ┣ 📜 okta.go
+       ┣ 📜 options.go
+       ┗ 📜 validate.go
+```
+
+### access.go
+
+Should contain an implementation for the accessor interface.
+
+```go
+type Accessor interface {
+	// Grant the access.
+	Grant(ctx context.Context, subject string, args []byte) error
+
+	// Revoke the access.
+	Revoke(ctx context.Context, subject string, args []byte) error
+}
+```
+
+It should also contain a struct `Args` always call this `Args` by convention.
+Args defines what is being requested when granting access. for example, in Okta, a groupId is requested. Args should contain all the
+parameters required by your provider implementation.
+
+Args must define both `json` and `jsonschema` tags there is some automation which happens whereby the Args struct is used to dynamically render part of the Create Access Rule form in the frontend. This allows flexibility when creating new providers because the frontend does not need to be updated to work with a new provider.
+
+```
+type Args struct {
+	GroupID string `json:"groupId" jsonschema:"title=Group"`
+}
+```
+
+The Grant and Revoke interface provides `args []byte` which will be a serialised json string. It can be unmarshalled into the Args type.
```

Comment:
```suggestion
The `Grant` and `Revoke` interface provides `args []byte` which will be a serialised json string. It can be unmarshalled into the Args type.
```